Strip spaces

fleetcaptain opened the pull request:
Hello, hope your day is going well! I think this fixes the issue with leading and trailing spaces in an email before hashing. I elected not to modify the stored email address, just to strip the spaces when hashing. If modifying the email value is desired let me know and I can implement that.

bsoyka commented on gravify/__init__.py:
```diff
@@ -38,7 +38,7 @@ def unsecure_url(self):
     @property
     def hash(self):
-        return hashlib.md5(self.email.encode("utf-8").lower()).hexdigest()
+        return hashlib.md5(self.email.strip(" ").encode("utf-8").lower()).hexdigest()
```
```suggestion
        return hashlib.md5(self.email.strip().encode("utf-8").lower()).hexdigest()
```
I think you should remove this string because without it, any whitespace characters will be removed. With it there however, only spaces are removed.

fleetcaptain replied:
I removed the space identifier from the strip(). I forgot there are tabs and other whitespace characters, good catch...

fleetcaptain commented:
Got it, thanks for the feedback! I'll work on this tonight or tomorrow and resubmit

 
 
  On Tue, Oct 1, 2019 at 17:39, Benjamin Soyka<notifications@github.com> wrote:   
@bsoyka requested changes on this pull request.

This looks good so far, but needs a couple changes.
   
   - See the suggestion I left on line 41.
   - The email also needs to be lowercased, as per the issue this PR is based on.
   - Please make a quick test case or two for this in /tests/test_main.py.

Thanks for your help so far!

In gravify/__init__.py:
> @@ -38,7 +38,7 @@ def unsecure_url(self):
 
     @property
     def hash(self):
-        return hashlib.md5(self.email.encode("utf-8").lower()).hexdigest()
+        return hashlib.md5(self.email.strip(" ").encode("utf-8").lower()).hexdigest()
⬇️ Suggested change-        return hashlib.md5(self.email.strip(" ").encode("utf-8").lower()).hexdigest()
+        return hashlib.md5(self.email.strip().encode("utf-8").lower()).hexdigest()

I think you should remove this string because without it, any whitespace characters will be removed. With it there however, only spaces are removed.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

bsoyka commented:
@fleetcaptain - Sounds good, and thanks again for your help!

bsoyka commented on tests/test_main.py:
```diff
@@ -15,6 +15,15 @@ def test_plain_email(self):
+    def test_hash_whitespace(self):
+        self.assertEqual(gravify.Gravatar("   \tbensoyka@icloud.com\n").hash, "7246821a7bf0b1b37794b39cb08ee052")
+
+    def test_hash_inner_spaces(self):
```
You probably don’t need this one, as it isn’t a valid email.

fleetcaptain replied:
Just have spaces in the test, no tabs or newlines?

fleetcaptain replied:
Oh, I see you called out the inner spaces test. Sorry, I was confused with a different test. I'll remove it

bsoyka replied:
No not that, I’m saying you don’t need `test_hash_inner_spaces`, as that’s not a valid email.

bsoyka replied:
Sounds good.

fleetcaptain replied:
Test case removed

fleetcaptain commented:
I believe I have updated my branch, if something looks off please let me know as I'm still getting learning my way around git... I removed the space character from the strip() call and implemented a few test cases.

It appears in the code that .lower() is already being called on the email address before it is hashed? The test case with upper case email yields the same hash as the test case with all lower case. Or should the email address be converted to lowercase before being stored? I could implement that if desired.

bsoyka commented:
Yep, you already had called `.lower()`, sorry about that.

Now that you mention it, it may be better to have the email changed before it is stored, that way the email validation checks the edited email. Let me know your thoughts on this one.

fleetcaptain commented:
I changed things so lower() and strip() are called on the email argument before getting stored.

fleetcaptain commented:
No problem, this was my first pull request to a third party repository, thank you for your patience and working with me!

bsoyka commented:
Of course! I’m glad to hear you’re contributing to other projects!

If you’re not participating already, I recommend you take part in this year’s [Hacktoberfest](https://hacktoberfest.digitalocean.com/).

And finally, I’d love it if you’d like to work on other “help wanted” issues on this project. I’m very happy to work with you, but if you’d like to simply move on, that’s your call. 😄

fleetcaptain commented:
Thanks, I'll take a look at the open issues and see what I can do!
